packages/bpk-component-infinite-scroll/src/withInfiniteScroll.js

```diff
@@ -0,0 +1,168 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import 'intersection-observer';
+
+import STYLES from './withInfiniteScroll.scss';
+
+const withInfiniteScroll = ComponentToExtend => {
+  class Infinitize extends Component {
+    constructor(props) {
+      super(props);
+
+      this.state = {
+        index: 0,
+        elementsToRender: [],
+        isListFinished: false,
+        showSeeMore: false,
+      };
+
+      this.onItemsFetch = props.onItemsFetch;
+      this.onScroll = props.onScroll;
+      this.handleIntersection = this.handleIntersection.bind(this);
```

Comment:
nit: this can be removed if you use the class property syntax + arrow function. e.g. `handleIntersection = async (entries) => {`

Reply:
changed